### PR TITLE
Support for the DataType Date in our graph. 

### PR DESCRIPTION
**Pull request by fppt**
- Note: The date is persisted as a string, so `order by` and `sort` operations on traversals will work on a string.

### CODE REVIEW

File: grakn-core/src/main/java/ai/grakn/concept/ResourceType.java [hunk 1]
```diff
@@ -169,12 +175,53 @@
...
+                LocalDateTime.class.getName(),
+                Schema.ConceptProperty.VALUE_DATE,
+                (d) -> d.atZone(ZoneId.systemDefault()).toInstant().toEpochMilli(),
+                (o) -> {
+                    if (o == null) return null;
+                    if (!(o instanceof Long)) {
+                        throw new InvalidConceptValueException(ErrorMessage.INVALID_DATATYPE.getMessage(o, Long.class.getName()));
+                    }
+                    return LocalDateTime.ofInstant(Instant.ofEpochMilli((long) o), ZoneId.systemDefault());
+                });
```
Comment (aelred) [inline grakn-core/src/main/java/ai/grakn/concept/ResourceType.java]: At this point, I would make these classes implementing an interface:

```
private static DateType implements DataType<LocalDateTime> {
    @Override
    public D getValue(Object o) {
        if (o == null) return null;
        if (!(o instanceof Long)) {
            throw new InvalidConceptValueException(ErrorMessage.INVALID_DATATYPE.getMessage(o, Long.class.getName()));
        }
        return LocalDateTime.ofInstant(Instant.ofEpochMilli((long) o), ZoneId.systemDefault());
    }
    ... etc ...
}
```
You can also give the interface a default implementation for the converter functions, to save restating it.

Comment (fppt) [inline grakn-core/src/main/java/ai/grakn/concept/ResourceType.java]: Can we make that it another PR? I just want to get the functionality in.

Comment (aelred) [inline grakn-core/src/main/java/ai/grakn/concept/ResourceType.java]: Yeah, I have no problem with that.
